# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: devtools/server/connectors/worker-connector.js
# Commit: 895cf08432c4
# Full Hash: 895cf08432c40fe371ff83e846cb161430c6be4b
# Author: Nicolas Chevobbe <nchevobbe@mozilla.com>
# Date: 2020-10-27 04:41:26
# Regressor Bug: 1633712
# File Overlap Count: 1
# Description:
#   Bug 1633712 - [devtools] Create WorkerTargets as soon as possible. r=ochameau,devtools-backward-compat-reviewers.
#   
#   This patch adds support for dedicated worker targets in the Watcher actor.
#   Shared and Service workers are not handled yet.
#   
# ==============================================================================

diff -r 2ce79a9ed2d5 -r 895cf08432c4 devtools/server/connectors/worker-connector.js
--- a/devtools/server/connectors/worker-connector.js	Mon Oct 26 12:52:26 2020 +0000
+++ b/devtools/server/connectors/worker-connector.js	Mon Oct 26 12:53:43 2020 +0000
@@ -21,11 +21,14 @@
  * @params {String} forwardingPrefix: The prefix that will be used to forward messages
  *                  to the DevToolsServer on the worker thread.
  * @params {Object} options: An option object that will be passed with the "connect" packet.
+ * @params {Object} options.watchedData: The watchedData object that will be passed to the
+ *                  worker target actor.
  */
 function connectToWorker(connection, dbg, forwardingPrefix, options) {
   return new Promise((resolve, reject) => {
-    if (dbg.isClosed) {
+    if (!DevToolsUtils.isWorkerDebuggerAlive(dbg)) {
       reject("closed");
+      return;
     }
 
     // Step 1: Ensure the worker debugger is initialized.
@@ -90,8 +93,9 @@
       dbg.addListener(listener);
     }
 
-    if (dbg.isClosed) {
+    if (!DevToolsUtils.isWorkerDebuggerAlive(dbg)) {
       reject("closed");
+      return;
     }
 
     // Step 2: Send a connect request to the worker debugger.
@@ -100,6 +104,11 @@
         type: "connect",
         forwardingPrefix,
         options,
+        workerDebuggerData: {
+          id: dbg.id,
+          type: dbg.type,
+          url: dbg.url,
+        },
       })
     );
 
@@ -134,7 +143,7 @@
         transport.ready();
         transport.hooks = {
           onClosed: () => {
-            if (!dbg.isClosed) {
+            if (DevToolsUtils.isWorkerDebuggerAlive(dbg)) {
               // If the worker happens to be shutting down while we are trying
               // to close the connection, there is a small interval during
               // which no more runnables can be dispatched to the worker, but